packages/block-library/src/search/edit.js

```diff
@@ -320,12 +322,32 @@ export default function SearchEdit( {
 	);
 
 	const getWrapperStyles = () => {
-		if ( 'button-inside' === buttonPosition && style?.border?.radius ) {
+		if ( 'button-inside' === buttonPosition && borderRadius ) {
 			// We have button inside wrapper and a border radius value to apply.
 			// Add default padding so we don't get "fat" corners.
 			//
-			// CSS calc() is used here to support non-pixel units. The inline
-			// style using calc() will only apply if both values have units.
+			// CSS calc() is used here to support non-pixel units.
+
+			if ( typeof borderRadius === 'object' ) {
+				// Individual corner border radii present.
+				const {
+					topLeft,
+					topRight,
+					bottomLeft,
+					bottomRight,
+				} = borderRadius;
+
+				return {
+					borderTopLeftRadius: `calc(${ topLeft } + ${ DEFAULT_INNER_PADDING })`,
```

Comment:
I just realised it's possible for this value to be `null`, so should these four lines be guarded behind a ternary check? E.g. `borderTopLeftRadius: topLeft ? `calc(${ topLeft } + ${ DEFAULT_INNER_PADDING })` : null` or something like that?

Reply:
I'd missed the possibility of unlinking the border radius control before a value was set, then adjusting only a single corner. I've updated this and just cleaned up the code a little further.